kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/channel/Coordinator.java

```diff
@@ -296,6 +301,28 @@ private void commitToTable(
     }
   }
 
+  private SnapshotAncestryValidator offsetValidator(
+      TableIdentifier tableIdentifier, Map<Integer, Long> expectedOffsets) {
+
+    return new SnapshotAncestryValidator() {
+      private Map<Integer, Long> lastCommittedOffsets;
+
+      @Override
+      public Boolean apply(Iterable<Snapshot> baseSnapshots) {
+        lastCommittedOffsets = lastCommittedOffsets(baseSnapshots);
+
+        return expectedOffsets.equals(lastCommittedOffsets);
+      }
+
+      @Override
+      public String errorMessage() {
+        return String.format(
+            "Latest offsets do not match expected offsets for this commit. Table: %s, Expected: %s, Last Committed: %s",
```

Comment:
This is less important in connector code vs core, but I like to have more direct error messages. For example, `"Cannot commit to %s, stale offsets:\nExpected: %s\nCommitted: %s"`